### PR TITLE
Fix a crash when renaming a child-object

### PR DESCRIPTION
**Pull request by D8H**
It adds tests for object renaming in events (layout and event-based object function)

### CODE REVIEW

File: Core/GDCore/IDE/WholeProjectRefactorer.cpp [hunk 1]
```diff
@@ -1628,19 +1628,18 @@ void WholeProjectRefactorer::ObjectOrGroupRemovedInEventsFunction(
 
 void WholeProjectRefactorer::ObjectOrGroupRenamedInEventsBasedObject(
     gd::Project& project,
-    gd::EventsBasedObject& eventsBasedObject,
     gd::ObjectsContainer& globalObjectsContainer,
```
Comment (4ian) [inline Core/GDCore/IDE/WholeProjectRefactorer.cpp:1631]: Open question: is the global objects container even making sense in an events based object?

File: Core/GDCore/IDE/WholeProjectRefactorer.cpp [hunk 2]
```diff
@@ -1628,19 +1628,18 @@ void WholeProjectRefactorer::ObjectOrGroupRemovedInEventsFunction(
...
-    gd::ObjectsContainer& objectsContainer,
+    gd::EventsBasedObject& eventsBasedObject,
     const gd::String& oldName,
     const gd::String& newName,
     bool isObjectGroup) {
   for (auto &functionUniquePtr : eventsBasedObject.GetEventsFunctions().GetInternalVector()) {
-    auto function = functionUniquePtr.get();
+    auto *function = functionUniquePtr.get();
```
Comment (4ian) [inline Core/GDCore/IDE/WholeProjectRefactorer.cpp:1637]: 😬

File: Core/tests/WholeProjectRefactorer.cpp [hunk 3]
```diff
@@ -106,14 +108,26 @@ const std::vector<const gd::EventsList *> GetEventsLists(gd::Project &project) {
   return eventLists;
 }
 
+const std::vector<const gd::ObjectsContainer *> GetObjectContainers(gd::Project &project) {
```
Comment (4ian) [inline Core/tests/WholeProjectRefactorer.cpp:111]: Should the name be more specific?

```suggestion
const std::vector<const gd::ObjectsContainer *> GetObjectContainersForMyOtherEventsBasedObject(gd::Project &project) {
```

**Review by 4ian — APPROVED** [review]

File: Core/GDCore/IDE/WholeProjectRefactorer.cpp (hunk 2 above)
Comment (D8H) [reply Core/GDCore/IDE/WholeProjectRefactorer.cpp:1637]: The `*` was implicit, so it has no impact.

File: Core/GDCore/IDE/WholeProjectRefactorer.cpp (hunk 1 above)
Comment (D8H) [reply Core/GDCore/IDE/WholeProjectRefactorer.cpp:1631]: No, the GUI hold an empty one to fill the parameter (like behaviors do).

File: Core/tests/WholeProjectRefactorer.cpp (hunk 3 above)
Comment (D8H) [reply Core/tests/WholeProjectRefactorer.cpp:111]: No, I should remove this method as I don't use it.
I though I would factorize tests for layout and ebo but it's not actually possible.

**Pull request merged**